app/src/main/java/com/android/sample/model/activity/ListActivitiesViewModel.kt

```diff
@@ -83,11 +87,56 @@ constructor(
   }
 
   fun addActivity(activity: Activity) {
-    repository.addActivity(activity, { getActivities() }, {})
+    repository.addActivity(
+        activity,
+        {
+          getActivities()
+          viewModelScope.launch {
+            try {
+              Firebase.auth.currentUser?.uid?.let { currentUserId ->
+                profilesRepository.getUser(
+                    userId = currentUserId,
+                    onSuccess = { currentUser ->
+                      if (currentUser != null) {
+                        App.getInstance()
+                            .scheduleNotification(
+                                activity = activity, isCreator = activity.creator == currentUser.id)
+                      }
+                    },
+                    onFailure = { e ->
+                      Log.e("ListActivitiesViewModel", "Failed to schedule notification", e)
+                    })
+              }
+            } catch (e: Exception) {
+              Log.e("ListActivitiesViewModel", "Error scheduling notification", e)
+            }
+          }
+        },
```

Comment:
This view model scrop launch does not belong here. It should be in the ProfileViewModel. Take a look at the fetchUserData, and try to see if adding your line 
`App.getInstance().scheduleNotification(
                                activity = activity, isCreator = activity.creator == currentUser.id
)` there is viable.

Reply:
I'm not sure I understand your suggestions. I wanted to do it here because the notification depends on the start time of the activity. Also, it works well with the current implementation so I'm not sure it's worth changing the whole logic.